### PR TITLE
Fix Titan23 Instances

### PR DESCRIPTION
**Pull request by kittobi1992**
This PR fixes the high running times and quality issues on the Titan23 instances. 

**Running Time Issue:** The instances contain very large hyperedges which causes high running times during (un)contractions. The changes introduces a proper threshold for removing very large hyperedges => max(0.01 * |V|, 50000) (same as in Mt-KaHyPar).

**Quality Issue:** The quality problem was due to the edge weight function in the community detection algorithm. We now use w(e)/|e| as edge weight for instances with density < 2 and with very large hyperedges. This change is not very elegant but fixes the issue for now.

### CODE REVIEW

**Comment by N-Maas** [comment]
Note that your change to IO conflicts with #167, where removing duplicate pins is moved to the input validation with a FastResetFlagArray

File: kahypar/partition/partitioner.h [hunk 1]
```diff
@@ -90,6 +90,7 @@ static inline void partition(Hypergraph& hypergraph, const Context& context) {
 class Partitioner {
  private:
   static constexpr bool debug = false;
+  static constexpr HypernodeID LARGE_HE_THRESHOLD = 1000;
```
Comment (SebastianSchlag) [inline kahypar/partition/partitioner.h:93]: Would it make sense to expose this as a config parameter?

File: kahypar/partition/partitioner.h [hunk 2]
```diff
@@ -177,7 +178,18 @@ inline void Partitioner::configurePreprocessing(const Hypergraph& hypergraph,
...
     } else {
-      context.preprocessing.community_detection.edge_weight = LouvainEdgeWeight::uniform;
+      bool contains_large_he = false;
+      for ( const HyperedgeID he : hypergraph.edges() ) {
+        if ( hypergraph.edgeSize(he) > LARGE_HE_THRESHOLD ) {
+          contains_large_he = true;
+          break;
+        }
+      }
+      if ( density < 2 && contains_large_he ) {
```
Comment (SebastianSchlag) [inline kahypar/partition/partitioner.h:188]: Would you mind adding a brief comment in the code here to explain the rationale behind this?

**Comment by SebastianSchlag** [comment]
> Note that your change to IO conflicts with #167, where removing duplicate pins is moved to the input validation with a FastResetFlagArray

Good point! @kittobi1992 could you coordinate this with @N-Maas ?

**Comment by SebastianSchlag** [comment]
@kittobi1992  it looks like some test cases are still failing.

File: kahypar/partition/partitioner.h (hunk 2 above)
Comment (kittobi1992) [reply kahypar/partition/partitioner.h:188]: done

File: kahypar/partition/partitioner.h (hunk 1 above)
Comment (SebastianSchlag) [reply kahypar/partition/partitioner.h:93]: Could you still add it as a tuning parameter please? That way, it at least becomes visible from the outside and doesn't get lost in somewhere in the code.

Comment (kittobi1992) [reply kahypar/partition/partitioner.h]: done

**Review by SebastianSchlag — APPROVED** [review]
Thanks for fixing this @kittobi1992! 👍

**Pull request merged**